plugin-taskbar/lxqtmasterpopup.cpp

```diff
@@ -0,0 +1,268 @@
+/* BEGIN_COMMON_COPYRIGHT_HEADER
+ * (c)LGPL2+
+ *
+ * LXDE-Qt - a lightweight, Qt based, desktop toolset
+ * http://razor-qt.org
+ * http://lxqt.org
+ *
+ * Copyright: 2011 Razor team
+ *            2014 LXQt team
+ * Authors:
+ *   Alexander Sokoloff <sokoloff.a@gmail.com>
+ *   Maciej Płaza <plaza.maciej@gmail.com>
+ *   Kuzma Shapran <kuzma.shapran@gmail.com>
+ *
+ * This program or library is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General
+ * Public License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * END_COMMON_COPYRIGHT_HEADER */
+
+#include "lxqtmasterpopup.h"
+#include "lxqtgrouppopup.h"
+#include "lxqttaskbutton.h"
+#include <QEnterEvent>
+#include <QDrag>
+#include <QMimeData>
+#include <QPropertyAnimation>
+#include <QLayout>
+#include "lxqttaskbar.h"
+#include <QStackedWidget>
+#include <QDebug>
+#include <QTimer>
+#include "../panel/ilxqtpanelplugin.h"
+#include <QPropertyAnimation>
+
+/************************************************
+    main purpose of this class is switching
+    between group menus and hiding itself
+    when mouse is not present
+ ************************************************/
+LxQtMasterPopup::LxQtMasterPopup(LxQtTaskBar *parent):
+    QFrame(parent),
+    mStackedWidget(new QStackedWidget(this)),
+    mCloseTimer(new QTimer(this)),
+    mFlags(0)
+{
+    mPosAnimation = new QPropertyAnimation(this,"pos",this);
+    mPosAnimation->setDuration(200);
+
+    mSizeAnimation = new QPropertyAnimation(this,"size",this);
+    mSizeAnimation->setDuration(200);
+
+    setWindowFlags(  Qt::CustomizeWindowHint | Qt::FramelessWindowHint | Qt::ToolTip);
+    setAttribute(Qt::WA_AlwaysShowToolTips);
+
+    QVBoxLayout * l = new QVBoxLayout(this);
+    l->addWidget(mStackedWidget);
+    l->setSpacing(0);
+    l->setMargin(0);
+
+    connect(mCloseTimer,SIGNAL(timeout()),this,SLOT(close()));
+    mCloseTimer->setSingleShot(true);
+    mCloseTimer->setInterval(400);
+}
+
+
+void LxQtMasterPopup::moveAnimated(const QPoint &newpos)
```

Comment:
This and resizeAnimated() are both unnecessary imo, they are just used in the one place and there's no reason they would be used anywhere else. Thoughts @luis-pereira ?